packages/did-client/src/did-client.ts

```diff
@@ -1,90 +1,308 @@
+interface VeridaSelfTransactionConfigPart  {
+    signer?: Signer
+    privateKey?: string
+}
+
+export interface DIDClientConfig {
+    // For signing doc
+    veridaPrivateKey: string
+
+    // vda-did resolver - only
+    identifier : string // DID
+    chainName? : string 
+    chainId? : string | number
+
+    // common to vda-did resolver & vda-did self transaction config
+    provider? : Provider
+    rpcUrl? : string
+    registry?: string
+    web3?: any
+
+    // vda-did
+    callType: CallType
+
+    // self or meta transaction configuration
+    web3Config: VeridaSelfTransactionConfigPart|VeridaMetaTransactionConfig
+}
+
+// Interfaces for saving DID Document
+interface VerificationMethodBase {
+    id: string
+    type: string
+    controller: string
+}
+
+interface Controller extends VerificationMethodBase {
+    blockchainAccountId: string
+}
+
+/**
+ * Interface to DIDClient instance
+ */
+export interface DIDClient {
+
+    authenticate(privateKey: string): void;
+
+    getDid(): string | undefined;
+
+    getPublicKey(): string;
+
+    saveDocument(document: DIDDocument): Promise<boolean>;
+
+    getDocument(): Promise<DIDDocument | null>
+    
+}
+
+/**
+ * Create instance of DIDClient interface. And load DIDDocument from chain
+ *  * 
+ * @param config : cofiguration for DIDClientImpl class
+ * @returns DIDClient 
+ */
+export async function createDIDClient(config:DIDClientConfig) : Promise<DIDClient> {
+    const didClient = new DIDClientImpl(config)
+
+    await didClient.loadDIDDocument()
+
+    return didClient
+}
+
+class DIDClientImpl implements DIDClient {
+
+    private config: DIDClientConfig
+    private identifier: string
+
+    // Verida Wallet Info
+    private veridaPrivateKey: Uint8Array
+    private veridaWallet: VeridaWallet
+
+    // private did?: string
+    
+    // Verida Doc
+    private didDoc? : DIDDocument
+
+
+    // vda-did resolver
+    private didResolver: Resolver
+    private vdaDid: VdaDID
+
+    constructor(config: DIDClientConfig) {
+        this.config = config
+
+        this.identifier = config.identifier
+
+        this.veridaWallet = new VeridaWallet(config.veridaPrivateKey)
+        this.veridaPrivateKey = this.veridaWallet.privateKeyBuffer
+
+        const vdaDidResolver = getResolver({
+            name: config.chainName,
+            provider: config.provider,
+            rpcUrl: config.rpcUrl,
+            registry: config.registry,
+            chainId: config.chainId,
+            web3: config.web3
+        })
+        
+        this.didResolver = new Resolver(vdaDidResolver)
+
+        const web3Config = config.callType === 'gasless' ?
+            <VeridaMetaTransactionConfig>config.web3Config :
+            <VeridaSelfTransactionConfig>{
+                ...<VeridaSelfTransactionConfigPart>config.web3Config,
+                provider: config.provider,
+                rpcUrl: config.rpcUrl,
+                web3: config.web3
+            }
+
+        this.vdaDid = new VdaDID({
```

Comment:
See below, this should only happen when the user "authenticate()"s.

Reply:
Done.